### PR TITLE
Add helper functions to add viper arguments and retrieve AuthConfig from them

### PR DESCRIPTION
**Pull request by JAORMX**
This creates the following helper functions:

* `RegisterViperOIDCFlags`: Registers common OIDC parameters to a command through viper.

* `GetAuthConfigFromFlags`: Builds an AuthConfig object from parsed viper arguments.

* `GetAuthConfigsFromFlags`: Builds multiple AuthConfig objects from parsed viper arguments.

* `ViperBindFlag`: wraps viper bindings and error checks.

These were created based on the constant re-definition and usage
in hollow-based projects.

### CODE REVIEW

**Comment by JAORMX** [comment]
cc @nicolerenee

File: ginjwt/helpers.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// A call to this would normally look as follows:
+//
+//		ginjwt.GetAuthConfigFromFlags(viper.GetViper())
+func GetAuthConfigFromFlags(v *viper.Viper) AuthConfig {
+	return AuthConfig{
+		Enabled:       v.GetBool("oidc.enabled"),
+		Audience:      v.GetString("oidc.audience"),
+		Issuer:        v.GetString("oidc.issuer"),
+		JWKSURI:       v.GetString("oidc.jwksuri"),
+		LogFields:     v.GetStringSlice("oidc.log"),
```
Comment (nicolerenee) [inline ginjwt/helpers.go:51]: You have log fields here but not in the idc struck above, otherwise I think this will be useful

Comment (JAORMX) [reply ginjwt/helpers.go:51]: I actually wanted to ask about this! I posted my question here: https://github.com/packethost/bender/pull/17 but never did get an answer. I saw this same pattern in archetype service and bender... where is this coming from usually? I never saw it defined in either code-bases.

File: ginjwt/helpers.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	ViperBindFlag("oidc.audience", cmd.Flags().Lookup("oidc-aud"))
+	cmd.Flags().String("oidc-issuer", "", "expected issuer of OIDC JWT")
+	ViperBindFlag("oidc.issuer", cmd.Flags().Lookup("oidc-issuer"))
+	cmd.Flags().String("oidc-jwksuri", "", "URI for JWKS listing for JWTs")
+	ViperBindFlag("oidc.jwksuri", cmd.Flags().Lookup("oidc-jwksuri"))
+	cmd.Flags().String("oidc-roles-claim", "claim", "field containing the permissions of an OIDC JWT")
+	ViperBindFlag("oidc.claims.roles", cmd.Flags().Lookup("oidc-roles-claim"))
+	cmd.Flags().String("oidc-username-claim", "", "additional fields to output in logs from the JWT token, ex (email)")
+	ViperBindFlag("oidc.claims.username", cmd.Flags().Lookup("oidc-username-claim"))
+}
```
Comment (JAORMX) [inline ginjwt/helpers.go:36]: @nicolerenee  I'm wondering about cases where there might be multiple issuers we need to validate against. (e.g. for the problem I'm trying to solve with https://github.com/metal-toolbox/hollow-toolbox/pull/4 ... I'm tempted on changing this to accept a slice and assume the order is correct. What do you think?

**Review by sfunkhouser — APPROVED** [review]

**Pull request merged**